src/frontend/Sider/Behandling/VedtakOgBeregning/Barnetilsyn/InnvilgeVedtak/InnvilgeBarnetilsyn.tsx

```diff
@@ -75,6 +78,11 @@ export const InnvilgeBarnetilsyn: React.FC<Props> = ({ lagretVedtak, barnMedOppf
     const [beregningsresultat, settBeregningsresultat] =
         useState(byggTomRessurs<BeregningsresultatTilsynBarn>());
 
+    useEffect(() => {
+        hentStønadsperioder(behandling.id);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [behandling.id]);
```

Comment:
Ryddigere om denne `useEffect` ligger i `useStønadsperioder` og så returnerer hooken bare ressursen?

Reply:
Og evt. gjøre samme for vilkårsperioder